### PR TITLE
SolutionRunner, Verifier, CppExecutor

### PR DESCRIPTION
**Pull request by HermanPlay**
1. basic verifier, which just scans both files, (may change to use of `diff` command).
2. SolutionRunner expanded to include CppExecutor and file comparing and optional file compiling
3. CppExecutor Which may compile files now, and execute them (without sandboxing for now)

### CODE REVIEW

File: solution/runner.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+	inputPath := fmt.Sprintf("%s/%s", baseDir, inputDir)
+	inputFiles, err := r.parseInputFiles(inputPath)
+	if err != nil {
+		log.Fatalf("error reading input directory. %s", err.Error())
+	}
+	userOutputDir := "user-output"
+	err = os.Mkdir(fmt.Sprintf("%s/%s", baseDir, userOutputDir), os.ModePerm)
+	if err != nil {
+		log.Fatalf("error creating user output directory. %s", err.Error())
```
Comment (Matios102) [inline solution/runner.go:107]: same here

File: solution/runner.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+			}
+		}
+	} else {
+		filePath = solutionFileName
+	}
+
+	inputPath := fmt.Sprintf("%s/%s", baseDir, inputDir)
+	inputFiles, err := r.parseInputFiles(inputPath)
+	if err != nil {
+		log.Fatalf("error reading input directory. %s", err.Error())
```
Comment (Matios102) [inline solution/runner.go:102]: return SolutionResult with Success: false and appropriate message

File: solution/runner.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+func (r *Runner) RunSolution(language LanguageConfig, baseDir string, solutionFileName string, inputDir string, outputDir string) SolutionResult {
+
+	// Init appropriate executor
+	var exec executor.Executor
+	var err error
+	switch language.Type {
+	// case PYTHON:
+	// 	exec, err = executor.NewPyExecutor(language.Version)
+	case CPP:
+		exec, err = executor.NewCppExecutor(language.Version)
```
Comment (Matios102) [inline solution/runner.go:69]: I guess we are not considering memory and time limits for now?

Comment (HermanPlay) [reply solution/runner.go:69]: We should, I just have not come with how to pass them effectively

**Pull request merged**